tests/chocolatey-tests/choco-install.Tests.ps1

```diff
@@ -0,0 +1,1378 @@
+        }
+
+        It "Exits with Success (0)" {
+            $Output.ExitCode | Should -Be 0
+        }
+
+        It "Should install the package in the lib directory" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest" | Should -Exist
+        }
+
+        It "Should still have the expected version installed" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec" | Should -Exist
+            [xml]$XML = Get-Content "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec"
+            $XML.package.metadata.version | Should -Be "1.0.0"
+        }
+
+        It "Should remove and readd the package files in the lib directory" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\tools\chocolateyInstall.ps1" | Should -Not -FileContentMatch "FileNotOverwritten"
+        }
+
+        It "Deletes the rollback files" {
+            "$env:ChocolateyInstall\lib-bkp\$PackageUnderTest" | Should -Not -Exist
+        }
+
+        It "Outputs a message indicating that it installed the package successfully" {
+            $Output.Lines | Should -Contain "Chocolatey installed 1/1 packages."
+        }
+    }
+
+    Context "Force Installing a Package that is already installed (that errors)" {
+        BeforeAll {
+            Restore-ChocolateyInstallSnapshot
+
+            $PackageUnderTest = "installpackage"
+
+            $null = Invoke-Choco install $PackageUnderTest --version 1.0.0 --confirm
+            "FileNotOverwritten" | Add-Content -Path "$env:ChocolateyInstall\lib\$PackageUnderTest\tools\chocolateyInstall.ps1"
+
+            $Output = Invoke-Choco install $PackageUnderTest --version 1.0.0 --force --confirm
+        }
+
+        It "Exits with Success (0)" {
+            $Output.ExitCode | Should -Be 0
+        }
+
+        It "Should install the package in the lib directory" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest" | Should -Exist
+        }
+
+        It "Should still have the expected version installed" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec" | Should -Exist
+            [xml]$XML = Get-Content "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec"
+            $XML.package.metadata.version | Should -Be "1.0.0"
+        }
+
+        It "Should remove and readd the package files in the lib directory" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\tools\chocolateyInstall.ps1" | Should -Not -FileContentMatch "FileNotOverwritten"
+        }
+
+        It "Deletes the rollback files" {
+            "$env:ChocolateyInstall\lib-bkp\$PackageUnderTest" | Should -Not -Exist
+        }
+
+        It "Outputs a message indicating that it installed the package successfully" {
+            $Output.Lines | Should -Contain "Chocolatey installed 1/1 packages."
+        }
+    }
+
+    Context "Force Installing a Package that is already installed (with a read/delete locked file)" {

```

Comment:
The wording of this test implies that the file is locked for the purposes of reading and deletion, but the test themselves appear to be verifying that choco _is_ in fact able to delete the file, because it was open with the FileShare modes that permit deletion?

Maybe something like:
```suggestion
    Context "Force Installing a Package that is already installed (with a locked file that permits read/delete)" {
```

Reply:
So this is the part that is thoroughly confusing to me. Reading the current documentation, the suggested change is what the documentation says it does. But looking at the behaviour difference between .Net 4.7.2 and 4.8 (#2690) it seems that the current documentation although it'll tell you is for 4.7.2, might actually be for 4.8...

Reply:
It might just be that we're relying somewhere on a default behaviour for a file delete operation that changed in the underlying .NET assemblies after 4.7.2, but yeah I agree it's not clear from the documentation.